### PR TITLE
feat: improve queries

### PR DESCRIPTION
**Pull request by amaanq**
Hello, I'm a treesitter maintainer for nvim-ts and a bunch of ts grammars, I use this plugin to enrich my (admittedly sparse) go development. I noticed the queries are a bit expensive and not as optimized as they could be for the end user, so I tidied it up and made some changes. Note that `#match?` is quite expensive and `#lua-match?` is much better.

Thanks,
Amaan

### CODE REVIEW

File: after/queries/go/injections.scm [hunk 1]
```diff
@@ -1,137 +1,134 @@
...
-            (identifier) @_id (#match? @_id ".*[Q|q]uery.*"))
-    right: (expression_list
-             (raw_string_literal) @sql (#offset! @sql 0 1 0 -1))
-)
-
+((const_spec
+  name: (identifier) @_const
+  value: (expression_list 
+           (raw_string_literal) @sql))
+ (#any-of? @_const "query" "Query")
```
Comment (ray-x) [inline after/queries/go/injections.scm:10]: The intention is to match any of 
userQuery = `select * from users where userID = ?`
query  = `select * fromxxx`
So maybe use either match or contains

File: after/queries/go/injections.scm [hunk 2]
```diff
@@ -1,137 +1,134 @@
-;; extends
+; extends
```
Comment (ray-x) [inline after/queries/go/injections.scm:1]: what is difference between `; extends` and `;; extends`
The help says `;; extends` but I saw someone use `; extends`

```
treesitter-query-modeline-extends
;; inherits: foo,bar
;; extends
;; extends
;;
;; inherits: baz
```

File: after/queries/go/injections.scm [hunk 3]
```diff
@@ -1,137 +1,134 @@
...
+           (raw_string_literal) @sql))
+ (#any-of? @_const "query" "Query")
+ (#offset! @sql 0 1 0 -1))
+
+((short_var_declaration
+  left: (expression_list
+          (identifier) @_var)
+  right: (expression_list
+           (raw_string_literal) @sql))
+  (#any-of? @_var "query" "Query")
```
Comment (ray-x) [inline after/queries/go/injections.scm:18]: same here, variable contains [q|Q] uery

**Review by ray-x — COMMENTED** [review]
thanks for the PR.
I have a few comments/questions. Please check if they make any sense.

File: after/queries/go/injections.scm (hunk 1 above)
Comment (amaanq) [reply after/queries/go/injections.scm:10]: I feel that query could be used for other things though, this type of match could be wrongly applied leading to weird highlights. but up to you

File: after/queries/go/injections.scm (hunk 2 above)
Comment (amaanq) [reply after/queries/go/injections.scm:1]: It is `; extends` not `;; extends`, the help should be updated..

**Comment by amaanq** [comment]
Side note, I don't exactly understand the point of the locals query..very weird naming convention there from what I can tell.

**Comment by amaanq** [comment]
> Here is an example
> 
> ```go
> const(
>   queryUser = "select * from users where user_id=?"
>   queryUserComments = "select * from user_comments where user_id=?"
>   queryUserID = "select user_id from users where user_name=?"
> )
> ```
> 
> Someone may prefer this instead
> 
> ```go
> const(
>   userQuery = "select * from users where user_id=?"
>   userCommentsQuery = "select * from user_comments where user_id=?"
> )
> ```

Right but query as a word doesn't just mean sql query, it has other meanings too and can be used for other things imo, but I guess for raw string literals only then it can be ok. Match is expensive so contains would be better, lmk if that's ok w/ you

**Comment by ray-x** [comment]
I agree `query`  is not a good way to detect SQL string.
maybe we combine it with keyword search inside strings 
; "ADD" "ADD CONSTRAINT" "ALL" "ALTER" "AND" "ASC" "COLUMN" "CONSTRAINT" "CREATE" "DATABASE" "DELETE" "DESC" "DISTINCT" "DROP" "EXISTS" "FOREIGN KEY" "FROM" "JOIN" "GROUP BY" "HAVING" "IN" "INDEX" "INSERT INTO" "LIKE" "LIMIT" "NOT" "NOT NULL" "OR" "ORDER BY" "PRIMARY KEY" "SELECT" "SET" "TABLE" "TRUNCATE TABLE" "UNION" "UNIQUE" "UPDATE" "VALUES" "WHERE"

**Comment by amaanq** [comment]
Yeah that's a better way to go about it, I'll edit this to include those and remove the var name matching.

**Review by ray-x — APPROVED** [review]
Thanks for cleaning it up!

**Pull request merged**